doc/troubleshooting.md

```diff
@@ -1,5 +1,36 @@
 # Troubleshooting
 
+## How email sending is triggered by content changes
+
+We attempt to send emails to subscribers when content changes if they have a subscription for a subscribable that matches.
+
+A change to a content item creates a `ContentChange` record, which has a `SubscriptionContent` that belongs to an `Email`. The `SubscriptionContent` is linked to a `Subscription`, which has a `Subscriber`, and belongs to a `Subscribable`. A `DeliveryAttempt` is made on the `Email`.
+
+![domain](https://github.com/alphagov/email-alert-api/blob/master/doc/domain.png?raw=true)
+
+## The digest run workflow
+
+A daily task (for the daily digest) or a Monday task (for the weekly digest) runs as a scheduled task in the `DigestInitiatorService`. This creates a `DigestRun`, then fetches the IDs of the subscribers and creates a `DigestRunSubscriber` record for each one, before enqueuing the digest run subscribers in the `DigestGenerationWorker`.
+
+This worker uses a `SubscriptionContentChangeQuery` to fetch subscriptions with content changes, and creates an `Email` using the `DigestEmailBuilder` and composed of `SubscriptionContent`.
+
+![digests](https://github.com/alphagov/email-alert-api/blob/master/doc/digests.png?raw=true)
+
+## How Email Alert API interacts with Notify
+
+email-alert-service makes a POST request which is handled by the email-alert-api `ContentChangeController`
+to create a `ContentChange` and enqueue a `ProcessContentChangeWorker`.
+
+This Sidekiq worker uses `SubscriptionMatcher` to find the affected `SubscriptionContent` and enqueues an `EmailGenerationWorker`.
+
+The [digest run workflow](#the-digest-run-workflow) is followed in another Sidekiq worker to enqueue the email.
+
+This last Sidekiq worker creates a `DeliveryAttempt` and sends the email to GOV.UK Notify. It changes its status to `sent_to_notify`.
```

Comment:
I believe the status is actually `sending`. https://github.com/alphagov/email-alert-api/blob/b01e51b82e153407035ae21fe6a04f7b341b8251/app/models/delivery_attempt.rb#L8